### DISCUSSION TITLE
Playing Higher Resolution Files

### DISCUSSION THREAD

**Discussion by judmarc** [jeffvli/feishin#69] [General]
I have some music in higher than CD resolution - 24/96 AIFF, etc. When I play these tracks from the computer with other software (for example, MinimServer controlled by a phone app), my DAC shows 24/96 input. When using feishin, it shows 44.1k input. Either feishin or Navidrome is resampling to the lower rate. Can you tell me which it is, and if it's feishin, is there now or planned to be something I can put in a config file or setting to keep the original 24/96 rate?

Thanks!

**Comment by jeffvli** [comment]
This should technically already be supported by using the `MPV parameters` option in the settings to add additional mpv parameters on startup.

If you check https://mpv.io/manual/stable/#audio, the following can be used:

`--audio-exclusive=yes` to set audio exclusive mode
`--audio-samplerate=<Hz value>` to set the sample rate

I'm also currently in the process of revamping the settings page to include commonly used mpv parameters (including the sample rate) for the next release!

![image](https://user-images.githubusercontent.com/42182408/229255610-5487985b-7922-44f6-813e-58eef49dc635.png)

**Reply by judmarc** [reply]
Thank you! From this and reading the mpv docs, it appears there's currently no option in mpv to keep the original sample rate per file when the sample rates of the originals vary and there are too many files to set each individually (in my case 10k+ files, not a huge number but definitely too many to set individually in a config file!). And the sample rates are all over the map. 

Looking forward to the next release so I can set a sample rate more conveniently. 🙂

**Reply by dweymouth** [reply]
I haven't tested with a DAC that shows sample rate, but I believe that as long as the `--gapless-audio` setting is not set to `yes`, then MPV should switch sample rates along with the source files. (`--gapless-audio=weak` is the behavior that you'd want and @jeffvli recently made it the default, since it allows mpv to close and re-open the audio device if there is a change in sample rate, and plays gaplessly if the consecutive files have the same rate)

**Reply by judmarc** [reply]
Thanks, will try it. Current mpv behavior under the weak setting says this, which makes me wonder whether rate might be converted anyway:

“Currently, the device is kept even if the sample format changes, but the sample formats are convertible.”

(Shrug. 🙂 Will check and see what the DAC says.)

**Comment by judmarc** [comment]
Still happily plays 24/96 album at 44.1k with --gapless-audio=weak. When I manually set rate to 96, play/pause button flashes rapidly back and forth between play and pause states and playback never moves off 0. Have tried with exclusive mode and without, same behavior. This is with Xubuntu and alpha 6.

The DAC is correctly indicating sample rate it receives, as with a couple of other apps original sample rate for albums (or upsampled rate if I've chosen to have the application do that) is shown on DAC.

**Reply by judmarc** [reply]
BTW, I can certainly send, for example, a 24/96 file for testing if you're interested. If this is more of a sidelight and not something that is an important goal for the player, then I don't want to take time away from priorities - we can just let the topic go.

**Reply by jeffvli** [reply]
Yeah if you can send a file that would be helpful.

**Reply by judmarc** [reply]
Thought of one more thing to try (files are AIFF, want to convert to FLAC to see if it makes any difference for me), then will send a couple.

**Comment by dweymouth** [comment]
I'll be curious to see if you can come to any resolution here. Other things to try might be setting audio exclusive mode (`--audio-exclusive=yes`)? And/or setting the --audio-device parameter explicitly. You can list audio devices by running on the command line `mpv --audio-device=help` and then take the hardware name (not the description) of your intended audio device and pass it as an option like `--audio-device=<deviceName>`

**Reply by judmarc** [reply]
Tried with and without exclusive mode, same result. Will look at the device parameter.
